packages/schemas/src/workflow-context.ts

```diff
@@ -0,0 +1,146 @@
+ * This step provides context about the task and previous iterations
+ *
+ * @param data - Context data including task info and previous iterations
+ * @returns A WorkflowAgentStep that can be injected
+ */
+export function buildPreContextStep(data: PreContextData): WorkflowAgentStep {
+  const { taskId, initialTask, previousIterations, currentIteration } = data;
+
+  // Build the iterations section if we have previous iterations
+  let iterationsSection = '';
+  if (previousIterations && previousIterations.length > 0) {
+    iterationsSection = '\n\n## Previous Iterations\n\n';
+    iterationsSection += previousIterations
+      .map(iter => {
+        let section = `### Iteration ${iter.number}\n\n`;
+
+        if (iter.title) {
+          section += `**Title:** ${iter.title}\n\n`;
+        }
+
+        if (iter.description) {
+          section += `**Description:** ${iter.description}\n\n`;
+        }
+
+        if (iter.plan) {
+          section += '**Plan:**\n```\n' + iter.plan + '\n```\n\n';
+        }
+
+        if (iter.changes) {
+          section += '**Changes:**\n```\n' + iter.changes + '\n```\n\n';
+        }
+
+        return section;
+      })
+      .join('\n');
+  }
+
+  // Build the current iteration section
+  let currentIterationSection = '';
+  if (currentIteration) {
+    currentIterationSection = `\n\n## Current Iteration (Iteration ${currentIteration.number})
+
+**This is the current iteration you are working on.**
+
+`;
+
+    if (currentIteration.title) {
+      currentIterationSection += `**Title:** ${currentIteration.title}\n\n`;
+    }
+
+    if (currentIteration.description) {
+      currentIterationSection += `**Description:** ${currentIteration.description}\n\n`;
+    }
+
+    if (currentIteration.plan) {
+      currentIterationSection +=
+        '**Plan:**\n```\n' + currentIteration.plan + '\n```\n\n';
+    }
+
+    if (currentIteration.changes) {
+      currentIterationSection +=
+        '**Changes from previous iteration:**\n```\n' +
+        currentIteration.changes +
+        '\n```\n';
+    }
+  }
+
+  // Build the prompt with all context information
+  const prompt = `You are receiving contextual information about this task execution.
+This information is provided to help you understand the task history and previous work.
+
+## Task Information
+
+**Task ID:** ${taskId}
+
+**Initial Task:**
+- **Title:** ${initialTask.title}
+- **Description:** ${initialTask.description}
+${iterationsSection}${currentIterationSection}
+
+## Your Role
+
+This context is provided for your awareness.`;
+
+  // Create the pre-context step
+  const step: WorkflowAgentStep = {
+    id: '__pre_context__',
+    type: 'agent',
+    name: 'Pre-Context Preparation',
+    prompt,
+    outputs: [
+      {
+        name: 'context_acknowledged',
+        description: 'Acknowledgment that context was received and understood',
+        type: 'string',
+      },
+    ],
+  };
```

Comment:
Based on the code, this `step` will be injected as the first step for new iterations. However, the step does generates a single string. The next steps won't be able to rely on any of the information processed in this step as agents start from scratch on every step. 

From the changes on `packages/agent/src/commands/run.ts`, I see we are adding a reference to the context files, so that's where each steps get the context. We can always generate a new file here with a summary and force a reference to that new file in the other steps. However, I would start with something simpler and just force the agents to read the `precontext` file you generated in `setup`.

Reply:
At first, this step was going to write on the filesystem, but I changed it to be plainly informative and I found it useful. You can see how the agent knows in which iteration it is, and the main goal of that iteration. However, we can get rid of it as it will consume tokens and the benefit is not huge.